views.py

```diff
@@ -0,0 +1,85 @@
+from datetime import datetime
+
+import requests
+from sqlalchemy import select
+from sqlalchemy.ext.asyncio import AsyncSession
+
+import schemas
+from database import models
+
+
+async def get_cities(db: AsyncSession):
+    query = await db.execute(
+        select(models.DBCity)
+    )
+    return query.scalars().all()
+
+
+async def post_city(city: schemas.CityCreate, db: AsyncSession):
+    new_city = models.DBCity(
+        name=city.name,
+        additional_info=city.additional_info,
+    )
+    db.add(new_city)
+    await db.commit()
+    await db.refresh(new_city)
+    return new_city
+
+
+async def delete_city(city_id: int, db: AsyncSession):
+    query = await db.execute(
+        select(models.DBCity)
+        .filter(models.DBCity.id == city_id)
+    )
+
+    found_city = query.scalars().first()
+
+    if found_city:
+        await db.delete(found_city)
+        await db.commit()
+        return True
+    return False
+
+
+async def fetch_data_temperature(db: AsyncSession):
+    query = await db.execute(
+        select(models.DBCity)
+    )
+    all_cities = query.scalars().all()
+
+    for city in all_cities:
+        url = f"http://api.weatherapi.com/v1/current.json?key=caef70c7d43a4a988c5163801242210&q={city.name}"
+        response = requests.get(url)
```

Comment:
The `fetch_data_temperature` function uses `requests.get` to fetch data from an external API. Since this is an asynchronous function, consider using an asynchronous HTTP client like `httpx` to avoid blocking the event loop.